segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/AsyncBaseChunkStorage.java

```diff
@@ -313,11 +313,11 @@ public AsyncBaseChunkStorage(Executor executor) {
     final public CompletableFuture<Integer> read(ChunkHandle handle, long fromOffset, int length, byte[] buffer, int bufferOffset) {
         Exceptions.checkNotClosed(this.closed.get(), this);
         // Validate parameters
-        Preconditions.checkArgument(null != handle, "handle");
+        Preconditions.checkArgument(null != handle, "handle must not be null");
         checkChunkName(handle.getChunkName());
-        Preconditions.checkArgument(null != buffer, "buffer");
+        Preconditions.checkArgument(null != buffer, "buffer must not be null");
         Preconditions.checkArgument(fromOffset >= 0, "fromOffset must be non-negative");
-        Preconditions.checkArgument(length >= 0 && length <= buffer.length, "length");
+        Preconditions.checkArgument(length >= 0 && length <= buffer.length, "length must be non-negative and must not exceed buffer");
```

Comment:
Please use `Exceptions.checkArrayRange`.  That will replace both this line and the next and add a couple more cases that it verifies.

Reply:
This needs to throw InvalidArgementException.

Reply:
Do not want to change method signature and tests for this change.